Makefile

```
@@ -455,72 +422,63 @@ docker-build-kubeadm-control-plane: ## Build the docker image for kubeadm contro
 	$(MAKE) set-manifest-image MANIFEST_IMG=$(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG)-$(ARCH) MANIFEST_TAG=$(TAG) TARGET_RESOURCE="./controlplane/kubeadm/config/default/manager_image_patch.yaml"
 	$(MAKE) set-manifest-pull-policy TARGET_RESOURCE="./controlplane/kubeadm/config/default/manager_pull_policy.yaml"
 
-.PHONY: docker-push
-docker-push: ## Push the docker images
-	docker push $(CONTROLLER_IMG)-$(ARCH):$(TAG)
-	docker push $(KUBEADM_BOOTSTRAP_CONTROLLER_IMG)-$(ARCH):$(TAG)
-	docker push $(KUBEADM_CONTROL_PLANE_CONTROLLER_IMG)-$(ARCH):$(TAG)
+.PHONY: e2e-framework
+e2e-framework: ## Builds the CAPI e2e framework
+	cd $(E2E_FRAMEWORK_DIR); go build ./...
 
 ## --------------------------------------
-## Docker — All ARCH
+## Testing
 ## --------------------------------------
 
-.PHONY: docker-build-all ## Build all the architecture docker images
-docker-build-all: $(addprefix docker-build-,$(ALL_ARCH))
+##@ test:
 
-docker-build-%:
-	$(MAKE) ARCH=$* docker-build
+ARTIFACTS ?= ${ROOT_DIR}/_artifacts
 
-.PHONY: docker-push-all ## Push all the architecture docker images
-docker-push-all: $(addprefix docker-push-,$(ALL_ARCH))
-	$(MAKE) docker-push-core-manifest
-	$(MAKE) docker-push-kubeadm-bootstrap-manifest
-	$(MAKE) docker-push-kubeadm-control-plane-manifest
+KUBEBUILDER_ASSETS ?= $(shell $(SETUP_ENVTEST) use --use-env -p path $(KUBEBUILDER_ENVTEST_KUBERNETES_VERSION))
 
-docker-push-%:
-	$(MAKE) ARCH=$* docker-push
+.PHONY: test
+test: $(SETUP_ENVTEST) ## Run all the tests
```

```suggestion
test: $(SETUP_ENVTEST) ## Run unit and integration tests
```

The description "Run all the tests" may be misleading as it's not actually running ALL tests but runs all unit and integration tests in the repo, ie. it doesn't run e2e tests.

https://cluster-api.sigs.k8s.io/developer/testing.html#running-unit-and-integration-tests

---

Done.